sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -5989,6 +5989,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_PRESERVE_LEGACY_COLUMN_ORDER =
+    buildConf("spark.sql.hive.preserveLegacyColumnOrder.enabled")
```

Comment:
Although it's not a mandatory, shall we have this at `HiveUtils.scala` file like the following?

https://github.com/apache/spark/blob/47e0888aca3618615ca8087038e887cc31a58aa8/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala#L125